### DISCUSSION TITLE
Where is autopilot in server side

### DISCUSSION THREAD

**Discussion by hitabm** [carla-simulator/carla#4621] [Q&A] [answered]
Hi, I need to find the class that implements autopilot behavior.
When running manual_control.py by pressing key **P** or by passing argument **-a**, autopilot gets enabled. I know that, this autopilot is different from automatic_control.py and it's controlling the ego vehicle from server side. Can any one please help me to find class related to this type of autopilot?
I have a doubt on **TrafficManager.cpp**.

**Comment by hitabm** [comment, marked as answer]
I found my own answer on documentation in [here](https://carla.readthedocs.io/en/latest/adv_traffic_manager/).

**Comment by SanggunLee** [comment]
the docs are there you found.

Doxygen will be useful.
Doxygen link is in the carla doc site.

I recommend to use cpp api not python api. python api is wrapper.
